cpp/src/io/statistics/column_statistics.cuh

```diff
@@ -101,28 +101,33 @@ struct calculate_group_statistics_functor {
                               !std::is_same_v<T, list_view>)>* = nullptr>
   __device__ void operator()(stats_state_s& s, uint32_t t)
   {
-    detail::storage_wrapper<block_size> storage(temp_storage);
-
-    using type_convert = detail::type_conversion<detail::conversion_map<IO, INT96>>;
-    using CT           = typename type_convert::template type<T>;
-    typed_statistics_chunk<CT, detail::statistics_type_category<T, IO>::include_aggregate> chunk;
-
-    for (uint32_t i = 0; i < s.group.num_rows; i += block_size) {
-      uint32_t r   = i + t;
-      uint32_t row = r + s.group.start_row;
-      if (r < s.group.num_rows) {
-        if (s.col.leaf_column->is_valid(row)) {
-          auto converted_value = type_convert::convert(get_element<T>(s, row));
-          chunk.reduce(converted_value);
-        } else {
-          chunk.null_count++;
+    // Temporarily disable stats writing for int96 timestamps
+    // TODO: https://github.com/rapidsai/cudf/issues/10438
+    if constexpr (!cudf::is_timestamp<T>() or IO != detail::io_file_format::PARQUET or
+                  INT96 != detail::is_int96_timestamp::YES) {
```

Comment:
I prefer returning early if condition is met to inverting the condition like this (less indentation, more direct logic). Something like
```
    if constexpr (cudf::is_timestamp<T>() and IO == detail::io_file_format::PARQUET and
                  INT96 == detail::is_int96_timestamp::YES) { return; }
```

Reply:
Ok. Sometimes this causes a compiler warning for declarations outside of the `if constexpr`. I will give it a try.